sources/lib/structures/ty.mli

```diff
@@ -135,7 +148,12 @@ val tsum : string -> string list -> t
 (** Create an enumeration type. [tsum name enums] creates an enumeration
     named [name], with constructors [enums]. *)
 
-val trecord : t list -> string -> (string * t) list -> t
+val t_adt :
+  ?body: ((string * (string * t) list) list) option ->
+  string -> t list ->
+  t
```

Comment:
Could you document (even rather quickly) the expected meaning of the heavily nested list type ?